plugin/src/main/java/org/opensearch/ml/action/memorycontainer/memory/TransportAddMemoriesAction.java

```diff
@@ -0,0 +1,1607 @@
+    final ClusterService clusterService;
+    final ConnectorAccessControlHelper connectorAccessControlHelper;
+    final MLFeatureEnabledSetting mlFeatureEnabledSetting;
+    final MLModelManager mlModelManager;
+    final MemoryContainerHelper memoryContainerHelper;
+    final MemoryEmbeddingHelper memoryEmbeddingHelper;
+
+    @Inject
+    public TransportAddMemoriesAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        SdkClient sdkClient,
+        NamedXContentRegistry xContentRegistry,
+        ClusterService clusterService,
+        ConnectorAccessControlHelper connectorAccessControlHelper,
+        MLFeatureEnabledSetting mlFeatureEnabledSetting,
+        MLModelManager mlModelManager,
+        MemoryContainerHelper memoryContainerHelper,
+        MemoryEmbeddingHelper memoryEmbeddingHelper
+    ) {
+        super(MLAddMemoriesAction.NAME, transportService, actionFilters, MLAddMemoriesRequest::new);
+        this.client = client;
+        this.sdkClient = sdkClient;
+        this.xContentRegistry = xContentRegistry;
+        this.clusterService = clusterService;
+        this.connectorAccessControlHelper = connectorAccessControlHelper;
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
+        this.mlModelManager = mlModelManager;
+        this.memoryContainerHelper = memoryContainerHelper;
+        this.memoryEmbeddingHelper = memoryEmbeddingHelper;
+    }
+
+    @Override
+    protected void doExecute(Task task, MLAddMemoriesRequest request, ActionListener<MLAddMemoriesResponse> actionListener) {
+        User user = RestActionUtils.getUserContext(client);
+        MLAddMemoriesInput input = request.getMlAddMemoryInput();
+
+        if (input == null) {
+            actionListener.onFailure(new IllegalArgumentException("Memory input is required"));
+            return;
+        }
+
+        String memoryContainerId = input.getMemoryContainerId();
+
+        if (StringUtils.isBlank(memoryContainerId)) {
+            actionListener.onFailure(new IllegalArgumentException("Memory container ID is required"));
+            return;
+        }
+
+        // Get memory container
+        memoryContainerHelper.getMemoryContainer(memoryContainerId, ActionListener.wrap(container -> {
+            // Check access
+            if (!memoryContainerHelper.checkMemoryContainerAccess(user, container)) {
+                actionListener
+                    .onFailure(
+                        new OpenSearchStatusException("User doesn't have permissions to add memory to this container", RestStatus.FORBIDDEN)
+                    );
+                return;
+            }
+
+            // Get memory index name
+            String indexName = memoryContainerHelper.getMemoryIndexName(container);
+            if (indexName == null) {
+                actionListener.onFailure(new IllegalStateException("Memory index not created for this container"));
```

Comment:
Is this error message accurate? Should it be "Memory index not set for this container"?